great_expectations/profile/basic_dataset_profiler.py

```diff
@@ -0,0 +1,117 @@
+from .base import DataSetProfiler
+
+
+class BasicDatasetProfiler(DataSetProfiler):
+    """A profiler inspired by the beloved pandas_profiling project
+    """
+
+    @classmethod
+    def _get_column_type_and_cardinality(cls, df, column):
+
+        if df.expect_column_values_to_be_of_type(column, "int")["success"]:
+            type_ = "int"
+
+        elif df.expect_column_values_to_be_of_type(column, "float")["success"]:
+            type_ = "float"
+
+        elif df.expect_column_values_to_be_of_type(column, "string")["success"]:
+            type_ = "string"
+
+        else:
+            type_ = "unknown"
+
+        unique = df.expect_column_unique_value_count_to_be_between(column, 0, None)[
+            'result']['observed_value']
+        pct_unique = df.expect_column_proportion_of_unique_values_to_be_between(
```

Comment:
I think setting explicit result_format is important here lest we collect entire columns as unexpected